File: Lecture 4/FinalCode - Java/app/src/main/java/com/inovaeg/androidtrainingpharos2020/dataStorage/SharedPrefActivity.java
package com.inovaeg.androidtrainingpharos2020.dataStorage;

import android.content.SharedPreferences;
import android.os.Bundle;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class SharedPrefActivity extends BaseStorageActivity {
    private SharedPreferences sharedPref;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        sharedPref = getSharedPreferences("SharedPref", MODE_PRIVATE);
    }

    @Override
    protected void saveNamesInSharedPref(List<String> names) {
        sharedPref.edit()
                .putStringSet(BaseStorageActivity.TYPE_KEY, new HashSet<>(names))
                .apply();
    }

    @Override
    protected List<String> restoreFromSharedPref() {
        Set<String> stringSet = sharedPref.getStringSet(BaseStorageActivity.TYPE_KEY, new HashSet<>());
        return new ArrayList<>(stringSet);

    }

}

File: Lecture 4/FinalCode - Kotlin/app/src/main/java/com/inovaeg/androidtrainingpharos2020/dataStorage/SharedPrefActivity.kt
package com.inovaeg.androidtrainingpharos2020.dataStorage

import android.content.SharedPreferences
import android.os.Bundle
import java.util.*

class SharedPrefActivity : BaseStorageActivity() {
    private lateinit var sharedPref: SharedPreferences
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        sharedPref = getSharedPreferences("SharedPref", MODE_PRIVATE)
    }

    override fun saveNamesInSharedPref(names: List<String>?) {
        names?.let {
            sharedPref.edit()
                    .putStringSet(BaseStorageActivity.Companion.TYPE_KEY, HashSet(names))
                    .apply()
        }
    }

    override fun restoreFromSharedPref(): MutableList<String>? {
        val stringSet = sharedPref.getStringSet(BaseStorageActivity.Companion.TYPE_KEY, HashSet())
                ?: HashSet()
        return ArrayList(stringSet)
    }
}